### DISCUSSION TITLE
PIXI-Spine - PIXI dependencies failing to be installed

### DISCUSSION THREAD

**Discussion by TyroTheFox** [pixijs/pixijs#8052] [Q&A] [answered]
Hi, 

It was suggested I post here after asking about my issue, [found here](https://github.com/pixijs/spine/issues/427), regarding my attempts to build typing declaration files for PIXI-Spine. Building normally through Rush appears to work, Types building doesn't. 

The short version appears to be that it's not installing parts of PIXI it expects to be there, according to the messages during 'rush install'. That causes issues, types don't happen. Looks like an issue finding PIXI stuff, really not familiar enough with what I'm looking at so I am asking for help. I stress; I downloaded the Master branch, followed the instructions as expected, made no modifications to the files and simply attempted to build the files.

Thanks for any input you can offer.

**Comment by TyroTheFox** [comment]
Anyone around? Another method of getting hold of typing for this library would be useful too.

**Comment by Dragonink** [comment, marked as answer]
I had the exact same problem as you, and I found a simple workaround.

The issue is that `@pixi-spine/base` needs `@pixi/settings`, `@pixi/runner` and `@pixi/ticker`, but they are not installed.
Let us just force them to be installed !
Add the three packages in the `peerDependencies` list of `packages/base/package.json` like this:
```diff
 "peerDependencies": {
     "@pixi/constants": "^6.1.0",
     "@pixi/core": "^6.1.0",
     "@pixi/display": "^6.1.0",
     "@pixi/graphics": "^6.1.0",
     "@pixi/math": "^6.1.0",
     "@pixi/mesh": "^6.1.0",
     "@pixi/mesh-extras": "^6.1.0",
     "@pixi/sprite": "^6.1.0",
     "@pixi/utils": "^6.1.0",
+    "@pixi/settings": "^6.1.0",
+    "@pixi/runner": "^6.1.0",
+    "@pixi/ticker": "^6.1.0"
 }
```
Now run `rush update` to update dependencies.
The missing packages should be installed now.

Finally run `npm run build`.
The `build` and `build:types` tasks should complete.

**Reply by TyroTheFox** [reply]
Fantastic, thanks. Trawling through the libraries myself, I was able to get it all to load up and run properly. Still not ideal to have to do this for something that touts 'Installing is as easy as running two commands, if you have Rush'. If no one else updates this, I might as well.
